depends/patches/qt/top_level_QtTopLevelHelpers.cmake

```diff
@@ -0,0 +1,565 @@
+# Populates $out_module_list with all subdirectories that have a CMakeLists.txt file
+function(qt_internal_find_modules out_module_list)
```

Comment:
Can you explain these patches? Seems like a lot of code copy-pasted out of Qt itself?

All `top_level_*` files are not "patches". They are indeed copy-pasted from https://github.com/qt/qt5 to reproduce Qt's full source infrastructure, which is required to configure multiple separate modules at once.

> Is this "normal" to have to do when trying to build Qt?

There are other alternatives:
1. Download the full source [archive](https://download.qt.io/official_releases/qt/6.7/6.7.2/single/).
2. Use the `qt-configure-module` tool.

I've tried the latter approach, and it makes the code much clunkier.

Reply:
> They are indeed copy-pasted from https://github.com/qt/qt5 to reproduce Qt's full source infrastructure,

It seems really odd that we'd need to copy-paste a bunch of code out of Qt 5, to compile Qt 6? Is this an approach supported by Qt? Is it guaranteed to keep working going forward, or just happens to work for now?

> Use the qt-configure-module tool.
> I've tried the latter approach, and it makes the code much clunkier.

Could you elaborate on "clunkier", as I assume at some point we are going to have to adopt this tooling in any case.

Reply:
> It seems really odd that we'd need to copy-paste a bunch of code out of Qt 5, to compile Qt 6?

The name of the repository is confusing. See the branches and tags for details. The `dev` branch is QT 6 :)

> Is this an approach supported by Qt? Is it guaranteed to keep working going forward, or just happens to work for now?

It is guaranteed to keep working going forward, even with additional modules for the QML GUI.

> Could you elaborate on "clunkier", as I assume at some point we are going to have to adopt this tooling in any case.

Instead of two packages, `qt` and `native_qt`, this approach assumes we have a separate package for each module: `qt`, `qttools`, `qttranslations`, `native_qt`, `native_qttools`, `native_qttranslations`. Additionally, the `*qttransaltions` module must have access to the `*qt` sources anyway.

Reply:
> The name of the repository is confusing. See the branches and tags for details. The dev branch is QT 6 :)

Ok. I guess in that case my confusing is just around having to need to copy-paste, and maintain code out of Qt 6, into separate patches, to then be able to compile Qt 6.

> It is guaranteed to keep working going forward, even with additional modules for the QML GUI.

Can you link to where this guarantee comes from? It'd be good to document all of this, if it's a less-standard way to build Qt.

Reply:
> Can you link to where this guarantee comes from? It'd be good to document all of this, if it's a less-standard way to build Qt.

Actually, it is the only documented way to configure Qt 6: https://doc.qt.io/qt-6/configure-options.html#configure-workflow.

Using the `qt-configure-module` tool is mentioned in the Qt's [blog](https://www.qt.io/blog/qt-6-build-system).

Reply:
> All `top_level_*` files are not "patches". They are indeed copy-pasted from https://github.com/qt/qt5 to reproduce Qt's full source infrastructure, which is required to configure multiple separate modules at once.

These files are fetched from the Qt repository as extra sources now.